model/datatable/DeliveriesActivityDatatable.php

```diff
@@ -42,19 +43,39 @@ class DeliveriesActivityDatatable implements DatatablePayload, ServiceLocatorAwa
     protected $request;
 
     /**
+     * @var DeliveryAssemblyService
+     */
+    protected $deliveryService;
+
+    /**
      * DeliveriesActivityDatatable constructor.
      */
     public function __construct()
     {
         $this->setServiceLocator(ServiceManager::getServiceManager());
         $this->request = DatatableRequest::fromGlobals();
+        $this->deliveryService = DeliveryAssemblyService::singleton();
     }
 
     public function getPayload()
     {
         /** @var ActivityMonitoringService $service */
         $service = $this->getServiceLocator()->get(ActivityMonitoringService::SERVICE_ID);
-        $data = $service->getStatesByDelivery();
+        $params = ['page' => $this->request->getPage(),
+            'rows' => $this->request->getRows(),
+            'sort_by' => $this->request->getSortBy(),
+            'sort_order' => $this->request->getSortOrder(),
+            'filters' => $this->request->getFilters()];
+
+        $limit = isset($params['rows']) ? $params['rows'] : 10;
+        $offset = isset($params['page']) ? ($params['page']-1) * $limit : 0;
+        $deliveries = $this->deliveryService->getRootClass()->getInstances(true, [
+            'order' => RDFS_LABEL,
+            'offset' => $offset,
+            'limit' => $limit
```

Comment:
You can't do pagination in this way. You always have to iterate through all existing deliveries. Otherwise ordering will work only for the current page.

Reply:
yes, I know, but it task become to complex with that improvement, so for now it will be ordering only on the current page but it will work on prod

Reply:
In this case remove pagination from this datatable at all. We can't add feature which does not work properly.

Reply:
We need this pagination because with 2000 deliveries it won't work too (it's a part of my fix)

Reply:
I could remove your sorting. I need to make quick fix for your code but not to create new feature

Reply:
Sorting by all columns was requested by ACT. Your PR is breaking this feature.